soh/soh/Enhancements/enemyrandomizer.cpp

```diff
@@ -0,0 +1,172 @@
+    { ACTOR_EN_TUBO_TRAP, 0 },  // Flying pot
+    { ACTOR_EN_FZ, 0 },         // Freezard
+    { ACTOR_EN_CLEAR_TAG, 1 },  // Arwing
+    { ACTOR_EN_WF, 0 },         // Wolfos (normal)
+    { ACTOR_EN_WF, 1 },         // Wolfos (white)
+    { ACTOR_EN_SKB, 1 },        // Stalchild (small)
+    { ACTOR_EN_SKB, 20 },       // Stalchild (big)
+    { ACTOR_EN_CROW, 0 }        // Guay
+
+    // Doesn't work {ACTOR_EN_POH, 0}, // Poe (Seems to rely on other objects?)
+    // Doesn't work {ACTOR_EN_POH, 2}, // Poe (composer Sharp) (Seems to rely on other objects?)
+    // Doesn't work {ACTOR_EN_POH, 3}, // Poe (composer Flat) (Seems to rely on other objects?)
+    // Doesn't work {ACTOR_EN_OKUTA, 0}, // Octorok (actor directly uses water box collision to handle hiding/popping up)
+    // Doesn't work {ACTOR_EN_REEBA, 0}, // Leever (reliant on surface and also normally used in tandem with a leever spawner, kills itself too quickly otherwise)
+};
+
+static int enemiesToRandomize[] = {
+    ACTOR_EN_FIREFLY,   // Keese (including fire/ice)
+    ACTOR_EN_TEST,      // Stalfos
+    ACTOR_EN_TITE,      // Tektite
+    ACTOR_EN_POH,       // Poe (normal, blue rupee, composers
+    ACTOR_EN_OKUTA,     // Octorok
+    ACTOR_EN_WALLMAS,   // Wallmaster
+    ACTOR_EN_REEBA,     // Leever
+    ACTOR_EN_PEEHAT,    // Flying Peahat, big one spawning larva, larva
+    ACTOR_EN_ZF,        // Lizalfos, dinolfos
+    ACTOR_EN_GOMA,      // Gohma larva (normal, eggs, gohma eggs)
+    ACTOR_EN_BUBBLE,    // Shabom (bubble)
+    ACTOR_EN_DODOJR,    // Baby Dodongo
+    ACTOR_EN_TORCH2,    // Dark Link
+    ACTOR_EN_BILI,      // Biri (small jellyfish)
+    ACTOR_EN_TP,        // Electric Tailparasan
+    ACTOR_EN_ST,        // Skulltula (normal, big, invisible)
+    ACTOR_EN_BW,        // Torch Slug
+    ACTOR_EN_EIYER,     // Stinger (land)
+    ACTOR_EN_MB,        // Moblins (Club, spear)
+    ACTOR_EN_DEKUBABA,  // Deku Baba (small, large)
+    ACTOR_EN_DEKUNUTS,  // Mad Scrub (single attack, triple attack)
+    ACTOR_EN_VALI,      // Bari (big jellyfish) (spawns very high up)
+    ACTOR_EN_BB,        // Bubble (flying skull enemy) (all colors)
+    ACTOR_EN_YUKABYUN,  // Flying Floor Tile
+    ACTOR_EN_VM,        // Beamos
+    ACTOR_EN_FLOORMAS,  // Floormaster
+    ACTOR_EN_RD,        // Redead, Gibdo
+    ACTOR_EN_SW,        // Skullwalltula
+    ACTOR_EN_FD,        // Flare Dancer
+    ACTOR_EN_KAREBABA,  // Withered Deku Baba
+    ACTOR_EN_RR,        // Like-Like
+    ACTOR_EN_NY,        // Spike (rolling enemy)
+    ACTOR_EN_IK,        // Iron Knuckle
+    ACTOR_EN_TUBO_TRAP, // Flying pot
+    ACTOR_EN_FZ,        // Freezard
+    ACTOR_EN_WEIYER,    // Stinger (Water)
+    ACTOR_EN_HINTNUTS,  // Hint deku scrubs
+    ACTOR_EN_WF,        // Wolfos
+    ACTOR_EN_SKB,       // Stalchild
+    ACTOR_EN_CROW       // Guay
+};
+
+extern "C" enemyEntry GetRandomizedEnemy(void) {
+    return randomizedEnemySpawnTable[rand() % RANDOMIZED_ENEMY_SPAWN_TABLE_SIZE];
```

Comment:
Would you give something like this a try? Probably a better way than just summing them up but this should allow it to be deterministic. (need to #include 3drando/random)
```suggestion
    int seed = 0;
    accumulate(gSaveContext.seedIcons, gSaveContext.seedIcons + 5, seed);
    Random_Init(seed);
    return randomizedEnemySpawnTable[Random(0, RANDOMIZED_ENEMY_SPAWN_TABLE_SIZE)];
```

Reply:
I'd rather focus on just finishing what's in it currently. The "chaos" version was my original intention, we can build off of it once this is done. I know you were just suggesting it to try it out, but keep in mind this has to work in vanilla saves as well.